### ISSUE TITLE
Update to latest JavaParser

### ISSUE THREAD

**Issue by matozoid** [moditect/moditect#24]
Always a good idea :-)

**Comment by gunnarmorling** [comment]
Indeed. Would you like to send a PR?

**Comment by matozoid** [comment]
I would have, but that would mean getting my system messed up with JDK 9 again.

**Comment by gunnarmorling** [comment]
Hehe :)

Btw. one of the goals I'm pursuing with ModiTect is to allow to add
addition of module descriptors (provided in pom.xml) to a project's output
JAR while being built on Java 8. Should be useful for libs that wish to
provide a module-info but still prefer to remain on Java 8 for their own
build.

**Comment by matozoid** [comment]
It would definitely be good for JavaParser, I tried upgrading everything but got stuck on some things. I will not try again within a year or so. It would be nice to have provide a module-info in the mean time.

Upgrading shouldn't be too hard, by the way. Maybe some things are deprecated and well described, or maybe something doesn't compile but it's just a signature that changed slightly. Most probably everything will still be fine.

**Comment by gunnarmorling** [comment]
Ok, cool, that's good to know. I'll try and do the update within the next
few days.

**Comment by gunnarmorling** [comment]
Hey @matozoid, just tried to upgrade to 3.5.0, and I'm seeing this:

```
com.github.javaparser.ParseProblemException: (line 1,col 1) Modules are not supported.

	at com.github.javaparser.JavaParser.simplifiedParse(JavaParser.java:346)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:314)
```

Is there any special method I need to call now to parse a module info file?

**Comment by matozoid** [comment]
Here you go: https://matozoid.github.io/2017/04/11/enable-java-9-support.html

**Comment by gunnarmorling** [comment]
Ah, yes, excellent! Thanks!

Am 28.10.2017 1:12 nachm. schrieb "Danny van Bruggen" <
notifications@github.com>:

**Comment by matozoid** [comment]
Sorry for underestimating the effort!

**Comment by gunnarmorling** [comment]
No worries, not that much work really, and thanks for your quick help! Just merged the upgrade 3.5.0.

That said, would you accept a pull request for adding ModiTect to the list of projects using JavaParser at http://javaparser.org/about.html?

**Comment by matozoid** [comment]
Of course, it's tough to keep an up to date list of active projects there!
